src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -507,17 +507,16 @@ private void OutputResultsHelper(IEnumerable<CommandInfo> results)
 
             if (UseFuzzyMatching)
             {
-                _commandScores = _commandScores
+                results = _commandScores
                     .Where(x => x.Score <= FuzzyMinimumDistance)
                     .OrderBy(static x => x.Score)
+                    .Select(static x => x.Command)
                     .ToList();
-                results = _commandScores.Select(static x => x.Command);
```

Comment:
By this change, the `results` and `_commandScores` are no longer one-to-one mappings.

Reply:
Ok, now I see what you were doing.  Have to make `_commandScores` not readonly.